packages-web/maps/src/components/GoogleMap.tsx

```diff
@@ -15,148 +15,139 @@ export interface GoogleMapsProps extends SharedProps {
     rotateControl: boolean;
 }
 
-export const GoogleMap = (props: GoogleMapsProps): ReactElement => {
+export function GoogleMap(props: GoogleMapsProps): ReactElement {
     const map = useRef<google.maps.Map>();
     const googleMapsRef = useRef<HTMLDivElement>(null);
-    const defaultCenterLocation: google.maps.LatLngLiteral = { lat: 51.906688, lng: 4.48837 };
-    let bounds!: google.maps.LatLngBounds;
-
     const markers = useRef<google.maps.Marker[]>([]); // Used to manage and remove markers from the map
 
     useEffect(() => {
         if (props.scriptsLoaded) {
-            createMap();
+            const mapOptions: google.maps.MapOptions = {
+                zoom: props.zoomLevel,
+                zoomControl: props.optionZoomControl,
+                scrollwheel: props.optionScroll,
+                draggable: props.optionDrag,
+                streetViewControl: props.optionStreetView,
+                mapTypeControl: props.mapTypeControl,
+                fullscreenControl: props.fullScreenControl,
+                rotateControl: props.rotateControl,
+                minZoom: 1,
+                maxZoom: 20,
+                styles: getGoogleMapsStyles(props.mapStyles)
+            };
+            if (googleMapsRef.current && !map.current) {
+                map.current = new google.maps.Map(googleMapsRef.current, mapOptions);
+            } else if (map.current) {
+                map.current.setOptions(mapOptions);
+            }
+            addMarkers(
+                map.current!,
+                markers.current,
+                props.locations,
+                props.currentLocation,
+                props.zoomLevel,
+                props.autoZoom
+            );
         }
     }, [props.scriptsLoaded]);
 
     useEffect(() => {
-        addMarkers();
-    }, [props.locations, props.currentLocation]);
-
-    const createMap = (): void => {
-        const mapOptions: google.maps.MapOptions = {
-            zoom: props.zoomLevel,
-            zoomControl: props.optionZoomControl,
-            scrollwheel: props.optionScroll,
-            draggable: props.optionDrag,
-            streetViewControl: props.optionStreetView,
-            mapTypeControl: props.mapTypeControl,
-            fullscreenControl: props.fullScreenControl,
-            rotateControl: props.rotateControl,
-            minZoom: 1,
-            maxZoom: 20,
-            styles: getGoogleMapsStyles(props.mapStyles)
-        };
-        if (googleMapsRef.current && !map.current) {
-            map.current = new google.maps.Map(googleMapsRef.current, mapOptions);
-            bounds = new google.maps.LatLngBounds().extend(defaultCenterLocation);
-        } else if (map.current) {
-            map.current.setOptions(mapOptions);
+        if (map.current) {
+            addMarkers(
```

Comment:
Couldn't it happen that we will `addMarkers` twice for initial render? If `scriptsLoaded` then first the first `useEffect` calls it and then the second.

Reply:
Logic rebuild using library for Google Api. Also removed addMarkers from useEffect with empty dependencies